phpunit/otgs-testcase.php

```diff
@@ -169,4 +169,63 @@ private function expectHookAdded( $action, $callback, $priority, $args = 1, $tim
 		$responder = \WP_Mock::onHookAdded( $action, $type )->with( $callback, $priority, $args );
 		$responder->perform( array( $intercept, 'intercepted' ) );
 	}
+
+	/**
+	 * Get an object protected property.
+	 *
+	 * @param object $object        Object.
+	 * @param string $property_name Property name.
+	 *
+	 * @return mixed
+	 *
+	 * @throws ReflectionException Reflection exception.
+	 */
+	private function get_property( $object, $property_name ) {
+		$reflection_class = new \ReflectionClass( $object );
+
+		$property = $reflection_class->getProperty( $property_name );
+		$property->setAccessible( true );
+		$value = $property->getValue( $object );
+		$property->setAccessible( false );
+
+		return $value;
+	}
+
+	/**
+	 * Set an object protected property.
+	 *
+	 * @param object $object        Object.
+	 * @param string $property_name Property name.
+	 * @param mixed  $value         Property vale.
+	 *
+	 * @throws ReflectionException Reflection exception.
+	 */
+	private function set_property( $object, $property_name, $value ) {
+		$reflection_class = new \ReflectionClass( $object );
+
+		$property = $reflection_class->getProperty( $property_name );
+		$property->setAccessible( true );
+		$property->setValue( $object, $value );
+		$property->setAccessible( false );
+	}
+
+	/**
+	 * Set an object protected method accessibility.
+	 *
+	 * @param object $object      Object.
+	 * @param string $method_name Property name.
+	 * @param bool   $accessible  Property vale.
+	 *
+	 * @return ReflectionMethod
+	 *
+	 * @throws ReflectionException Reflection exception.
+	 */
+	private function set_method_accessibility( $object, $method_name, $accessible = 'true' ) {
+		$reflection_class = new \ReflectionClass( $object );
+
+		$method = $reflection_class->getMethod( $method_name );
+		$method->setAccessible( $accessible );
+
+		return $method;
+	}
```

Comment:
@Igor-WPML are these methods needed to solve a specific problem?

I don't like the idea to access protected properties. We are supposed to only test the public API of a class.

Anyway, if we need these changes, these methods must be renamed.

Something like that:

- `get_property` -> `get_protected_property`
- `set_property` -> `set_protected_property`

Reply:
@andreasciamanna I understand that we should test API, but sometimes it is quite complicated to write tests not manipulating with protected properties. Please look at wpmlcore-6999. Also, I have already used one of these methods in `wp-content/plugins/sitepress-multilingual-cms/tests/phpunit/tests/inc/test-icl-cache.php`. I will remove it there and use just added to OTGS_TestCase.

Reply:
@andreasciamanna I have renamed methods as proposed.

Reply:
I would prefer to avoid adding new features to this library.
The reasons are:

1. It often causes issues with different versions of PHP (some time ago, Toolset had to create a separate branch to make it compatible with more recent versions of PHP)
2. In most of the cases, we should be able to write our tests (at least the new ones) without extending `OTGS_TestCase`.


Anyway, if this change is pertinent with the fixes for PHP 7.4, let's go ahead.